### PR TITLE
feat(git-utils): extract the diff content

### PR DESCRIPTION
**Pull request by tupe12334**
Close: #5204 

## PR Details

This pull request extract the diff between the current state of the amplication branch to the latest amplication commit

## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-git-utils/src/git/git.service.ts [hunk 1]
```diff
@@ -144,6 +154,34 @@ export class GitClientService {
...
+      owner,
+      repositoryName,
+    });
+
+    const latestCommit = commitsList[0];
+
+    if (!latestCommit) {
+      throw new Error(
+        "Didn't found a commit that has been created by amplication"
+      );
```
Comment (abrl91) [inline packages/amplication-git-utils/src/git/git.service.ts:177]:
```suggestion
      throw new Error(
        "Didn't find a commit that has been created by Amplication"
      );
```

File: packages/amplication-git-utils/src/git/git.service.ts [hunk 2]
```diff
@@ -109,6 +111,14 @@ export class GitClientService {
         branchName,
         clone: localRepository,
       });
+
+      const diffContent = await this.preCommitProcess({
```
Comment (abrl91) [inline packages/amplication-git-utils/src/git/git.service.ts:115]: Where this variable is being used?

File: packages/amplication-git-utils/src/git/git.service.ts [hunk 3]
```diff
@@ -144,6 +154,34 @@ export class GitClientService {
...
+      branchName,
+      owner,
+      repositoryName,
+    });
+
+    const latestCommit = commitsList[0];
+
+    if (!latestCommit) {
+      throw new Error(
+        "Didn't found a commit that has been created by amplication"
```
Comment (abrl91) [inline packages/amplication-git-utils/src/git/git.service.ts:176]:
```suggestion
      throw new Error(
        "Didn't find a commit that has been created by amplication"
```

File: packages/amplication-git-utils/src/git/git.service.ts (hunk 1 above)
Comment (abrl91) [reply packages/amplication-git-utils/src/git/git.service.ts:177]: maybe use this error message as a const because we use it in more than one place

**Review by EugeneTseitlin — APPROVED** [review]

**Review by abrl91 — APPROVED** [review]

**Pull request merged**